src/NServiceBus.Persistence.CosmosDB.AzureStorageSagaExporter/ToolVersion.cs

```diff
@@ -36,7 +36,7 @@ public static async Task<bool> CheckIsLatestVersion(ILogger logger)
                 var latest = versions.OrderByDescending(pkg => pkg.Version).FirstOrDefault();
                 var current = new NuGetVersion(GitVersionInformation.NuGetVersionV2);
 
-                if (latest > current)
+                if (latest > current || ignoreUpdates)
```

Comment:
Is this logic even correct?

If `ignoreUpdates` has a value the flag will be `true`, if `latest > current` is true due to the conditional logic OR operator the right hand side will not be evaluated. Therefore we return false and then the passing the flag doesn't even help

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-or-operator-

> The result of x || y is true if either x or y evaluates to true. Otherwise, the result is false. If x evaluates to true, y is not evaluated.

Reply:
Raised https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/61